sycamore/transforms/merge_elements.py

```diff
@@ -0,0 +1,128 @@
+from abc import ABC, abstractmethod
+
+from ray.data import Dataset, ActorPoolStrategy
+
+from sycamore.data import Document, Element, BoundingBox
+from sycamore.plan_nodes import NonCPUUser, NonGPUUser, Transform, Node
+from sycamore.transforms.map import generate_map_class_from_callable
+from sycamore.functions.tokenizer import Tokenizer
+
+from typing import Tuple
+
+
+class ElementMerger(ABC):
```

Comment:
This is such a generic interface, it's effectively equivalent to just passing the merger to the map function, right? The logic still makes sense, I guess I'm just wondering if there is anything more to put in the abstraction. Is it worth exposing an interface that just merges to elements, and then we are responsible for calling it? Is there any logic you expect to be common to all mergers?

Reply:
I considered exposing a `should_merge` and a `merge_two` interface kind of deal, and then have built in logic that does the reduce operation

Reply:
might also be worth providing a `preprocess_elements` and `postprocess_elements` interface?